src/DataStructures/Mesh.hpp

```diff
@@ -0,0 +1,149 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the class template Mesh.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+#include "DataStructures/Index.hpp"
+#include "Utilities/Requires.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+namespace Spectral {
+enum class Basis;
+enum class Quadrature;
+}  // namespace Spectral
+/// \endcond
+
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief The computational grid. Simulated fields are represented by
```

Comment:
"Holds the number of grid points, basis, and quadrature in each direction of the computational grid"